### PR TITLE
Refer to feature policy spec

### PR DESCRIPTION
**Pull request by clelland**
This change defines Fullscreen as a policy-controlled feature,
referencing the feature policy spec. It makes use of the currently-
monkey-patched "Allowed to use" algorithm in HTML to allow the
document's policy to control whether an element is permitted to go
fullscreen.

Fixes: #106

***
<a href="https://whatpr.org/fullscreen/107.html" title="Last updated on Jul 6, 2018, 6:54 PM GMT (19ba697)">Preview</a> | <a href="https://whatpr.org/fullscreen/107/6e49fbc...19ba697.html" title="Last updated on Jul 6, 2018, 6:54 PM GMT (19ba697)">Diff</a>

### CODE REVIEW

**Comment by foolip** [comment]
@clelland, I accidentally lost this in my inbox way back, sorry! I've rebased it and tried some tweaks locally. Sent https://github.com/WICG/feature-policy/pull/179 to export two definitions needed, can you merge that?

**Comment by clelland** [comment]
Merged! (I thought those were exported already; thanks for fixing that)

File: fullscreen.bs [hunk 1]
```diff
@@ -629,6 +629,23 @@ iframe:fullscreen {
...
+<p>The Fullscreen API is a policy-controlled feature, as defined by [[!FEATURE-POLICY]].
+
+<ul>
+  <li>The feature name for the Fullscreen API is "fullscreen".
+  <li>The <a>default allowlist</a> for the Fullscreen API is '<code>self</code>'.
+</ul>
+
+<p>A <a>document</a>'s <a>feature policy</a> determines whether any content in that document is allowed to
+go fullscreen. If disabled in any document, no content in the document will be <a>allowed to use</a>
+fullscreen. If enabled, content in the document will be <a>allowed to use</a> fullscreen.
```
[the remark this answers is not in the document]
Comment (clelland) [reply fullscreen.bs]: Yes, I have a PR at https://github.com/whatwg/html/pull/3287 which attempts to do just that. Working on a couple of final integration points before that can (hopefully) be merged.

[the remark this answers is not in the document]
Comment (foolip) [reply fullscreen.bs]: Should this PR be put on pause for a bit while that is finalized?

[the remark this answers is not in the document]
Comment (clelland) [reply fullscreen.bs]: Now that that has landed, I've updated the PR to be more in line with it.

**Comment by clelland** [comment]
@foolip, can you take a look? I've updated the wording in the PR to be more like what was used in XHR, and to work with the changes to HTML. I also changed what is really a non-normative description of the way that `allow` and `allowfullscreen` work into a `<div class="note">`.

**Comment by clelland** [comment]
Yes, fullscreen is pretty well tested, being one of the first features that Chrome shipped with -- these tests all check the behaviour of `fullscreen *` in policies:

* [feature-policy-frame-policy-allowed-for-all.https.sub.html](https://wpt.fyi/results/feature-policy/feature-policy-frame-policy-allowed-for-all.https.sub.html)
* [feature-policy-header-policy-allowed-for-all.https.sub.html](https://wpt.fyi/results/feature-policy/feature-policy-header-policy-allowed-for-all.https.sub.html)
* [feature-policy-nested-header-policy-allowed-for-all.https.sub.html](https://wpt.fyi/results/feature-policy/feature-policy-nested-header-policy-allowed-for-all.https.sub.html)
* [feature-policy-nested-header-policy-allowed-for-self.https.sub.html](https://wpt.fyi/results/feature-policy/feature-policy-nested-header-policy-allowed-for-self.https.sub.html)

They do all rely on the [`document.policy`](https://drafts.featurepolicy.rocks/add-introspection/#introspection) object which isn't shipped yet, though, so it may be worthwhile to add a couple of extra tests for that specifically.

(In general, I'd like to get to a point where we can test the feature policy mechanism itself separately from its integration with each feature -- with the goal being that if you can test that the feature works correctly when enabled and when disabled, then the wpt/feature-policy tests will ensure that all of the different possible combinations of frames and policies work to get the right {enabled,disabled} state down to the frame. For now, though, I don't mind adding tests for specific combinations of policy-structure+feature to increase test coverage.)

**Comment by foolip** [comment]
Ah, so those tests all fail. But they do cover the bits changed in this PR, so I'll go ahead and merge now. Thanks!

**Comment by clelland** [comment]
Yes -- they pass in Chrome with --enable-experimental-web-platform-features, but not without.
Thanks!

**Comment by foolip** [comment]
Ah, yes, it works!
https://wpt.fyi/results/feature-policy/feature-policy-frame-policy-allowed-for-all.https.sub.html?label=experimental

**Pull request merged**